Tuya 3.1 Dimmers Online but not responding to commands or updates

I have two Tuya light dimmers.  They are 3.1 firmware.  They didn't come online initially, but when I manually entered the ip addresses they did.  But they are not responsive to commands sent through the channels and generate errors when the devices are controlled through the Tuya SmartLife App.  

openHAB 3.2 Release Version with 3.2.10 Smarthome/J Tuya Binding

Log Info Level
`2022-02-17 16:01:06.491 [WARN ] [inding.tuya.internal.util.CryptoUtil] - Decryption of MQ failed: Input length not multiple of 16 bytes
`
Log TRACE showing heartbeat communication:
`2022-02-17 15:44:57.918 [INFO ] [a.internal.handler.TuyaDeviceHandler] - Configuring IP address 'DeviceInfo{ip='192.168.1.245', version='3.1'}' for thing 'tuya:tuyaDevice:42434288807d3a372fd5'.
2022-02-17 16:02:45.928 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Sending ping
2022-02-17 16:02:45.929 [DEBUG] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending HEART_BEAT, payload {devId=42434288807d3a372fd5, uid=42434288807d3a372fd5, t=1645135365, dps=, gwId=42434288807d3a372fd5}
2022-02-17 16:02:45.930 [TRACE] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending encoded '000055AA0000006C000000090000007B7B226465764964223A223432343334323838383037643361333732666435222C22756964223A223432343334323838383037643361333732666435222C2274223A313634353133353336352C22647073223A22222C2267774964223A223432343334323838383037643361333732666435227D0BBF6FC50000AA55'
2022-02-17 16:02:45.939 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000090000000C00000000B051AB030000AA55'
2022-02-17 16:02:45.940 [DEBUG] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-17 16:02:45.941 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Received pong
2022-02-17 16:02:55.933 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Sending ping
2022-02-17 16:02:55.934 [DEBUG] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending HEART_BEAT, payload {devId=42434288807d3a372fd5, uid=42434288807d3a372fd5, t=1645135375, dps=, gwId=42434288807d3a372fd5}
2022-02-17 16:02:55.935 [TRACE] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending encoded '000055AA0000006D000000090000007B7B226465764964223A223432343334323838383037643361333732666435222C22756964223A223432343334323838383037643361333732666435222C2274223A313634353133353337352C22647073223A22222C2267774964223A223432343334323838383037643361333732666435227DDE8011BE0000AA55'
2022-02-17 16:02:55.950 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000090000000C00000000B051AB030000AA55'
2022-02-17 16:02:55.952 [DEBUG] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-17 16:02:55.952 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Received pong
2022-02-17 16:03:05.938 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Sending ping
2022-02-17 16:03:05.940 [DEBUG] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending HEART_BEAT, payload {devId=42434288807d3a372fd5, uid=42434288807d3a372fd5, t=1645135385, dps=, gwId=42434288807d3a372fd5}
2022-02-17 16:03:05.941 [TRACE] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending encoded '000055AA0000006E000000090000007B7B226465764964223A223432343334323838383037643361333732666435222C22756964223A223432343334323838383037643361333732666435222C2274223A313634353133353338352C22647073223A22222C2267774964223A223432343334323838383037643361333732666435227D9A5EB8BE0000AA55'
2022-02-17 16:03:05.948 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000090000000C00000000B051AB030000AA55'
2022-02-17 16:03:05.950 [DEBUG] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-17 16:03:05.951 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Received pong
2022-02-17 16:03:15.945 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Sending ping
2022-02-17 16:03:15.946 [DEBUG] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending HEART_BEAT, payload {devId=42434288807d3a372fd5, uid=42434288807d3a372fd5, t=1645135395, dps=, gwId=42434288807d3a372fd5}
2022-02-17 16:03:15.946 [TRACE] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending encoded '000055AA0000006F000000090000007B7B226465764964223A223432343334323838383037643361333732666435222C22756964223A223432343334323838383037643361333732666435222C2274223A313634353133353339352C22647073223A22222C2267774964223A223432343334323838383037643361333732666435227D4F61C6C50000AA55'
2022-02-17 16:03:15.958 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000090000000C00000000B051AB030000AA55'
2022-02-17 16:03:15.959 [DEBUG] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-17 16:03:15.960 [TRACE] [rnal.local.handlers.HeartbeatHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Received pong`


Log TRACE level when changed on SmartLife
`2022-02-17 16:06:19.990 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000080000008B00000000332E31623162626331663736653237643162306A5074684C6356484F4963475A6A4B57784F52553567514863'
2022-02-17 16:06:19.991 [TRACE] [.internal.local.handlers.TuyaDecoder] - Did not receive enough bytes from '42434288807d3a372fd5', exiting early
2022-02-17 16:06:19.992 [TRACE] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received encoded '000055AA00000000000000080000008B00000000332E31623162626331663736653237643162306A5074684C6356484F4963475A6A4B57784F52553567514863776A745837616B376467376A7856647262585369492F65474C52586B3937624D6978487558496C525A536F41536754475251336B396B526A717A733979497258337A426674706B47663768614F6B6E6673593D3CF6B4D80000AA55'
2022-02-17 16:06:19.993 [WARN ] [inding.tuya.internal.util.CryptoUtil] - Decryption of MQ failed: Input length not multiple of 16 bytes
`

Configuration from UI Code

`UID: tuya:tuyaDevice:42434288807d3a372fd5
label: Media Lights (Tuya)
thingTypeUID: tuya:tuyaDevice
configuration:
  protocol: "3.1"
  productId: IaA6oFXfDu3tm1Pc
  deviceId: 42434288807d3a372fd5
  ip: 192.168.1.245
  localKey: f93875122317ca44
channels:
  - id: bright_value
    channelTypeUID: tuya:dimmer
    label: bright_value
    description: null
    configuration:
      min: 11
      dp: 2
      max: 255
      dp2: 1
  - id: TuyaMediaDimmerPowerSwitch
    channelTypeUID: tuya:switch
    label: Media Room Dimmer Power Switch
    description: ""
    configuration:
      dp: 1
`

Disabled and reenabled the device

`2022-02-17 16:09:12.869 [DEBUG] [al.local.handlers.TuyaMessageHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Connection terminated.
2022-02-17 16:09:14.166 [DEBUG] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP, content='{"ip":"192.168.1.245","gwId":"42434288807d3a372fd5","active":2,"ability":0,"mode":0,"encrypt":true,"productKey":"IaA6oFXfDu3tm1Pc","version":"3.1"}'}
2022-02-17 16:09:16.794 [INFO ] [a.internal.handler.TuyaDeviceHandler] - Configuring IP address 'DeviceInfo{ip='192.168.1.245', version='3.1'}' for thing 'tuya:tuyaDevice:42434288807d3a372fd5'.
2022-02-17 16:09:16.805 [DEBUG] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending DP_QUERY, payload {devId=42434288807d3a372fd5, uid=42434288807d3a372fd5, t=1645135756, dps=null, gwId=42434288807d3a372fd5}
2022-02-17 16:09:16.806 [TRACE] [.internal.local.handlers.TuyaEncoder] - 42434288807d3a372fd5/192.168.1.245:6668: Sending encoded '000055AA000000010000000A0000007B7B226465764964223A223432343334323838383037643361333732666435222C22756964223A223432343334323838383037643361333732666435222C2274223A313634353133353735362C22647073223A7B7D2C2267774964223A223432343334323838383037643361333732666435227D9A7E37B70000AA55'
2022-02-17 16:09:16.806 [DEBUG] [al.local.handlers.TuyaMessageHandler] - 42434288807d3a372fd5/192.168.1.245:6668: Connection established.
`

Thanks.  The development version loaded and the Things came online.  But I am seeing this in the log:
`2022-02-18 09:22:01.894 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.148', deviceId='ebf3cb07f3574a9688z4oy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
==> /var/log/openhab/events.log <==
2022-02-18 09:22:02.101 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:52.683 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.246', deviceId='eb755c84e755bbd761ymwj', active=2, ability=0, mode=0, encrypt=true, productKey='keyyj3fy8x98arty', version='3.3'}'}
2022-02-18 09:22:02.103 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:53.203 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.226', deviceId='ebb442e79ea9c02fb35xs8', active=2, ability=0, mode=0, encrypt=true, productKey='keytg5kq8gvkv9dh', version='3.3'}'}
2022-02-18 09:22:02.104 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:53.697 [ERROR] [.internal.local.handlers.TuyaDecoder] - 36802558840d8e95ef5d/192.168.1.244:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.107 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:53.725 [ERROR] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a3734e6/192.168.1.229:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.109 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:53.896 [ERROR] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.111 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:53.960 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.180', deviceId='eb3c82adf880a32bbe5ivy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:02.113 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:54.053 [ERROR] [.internal.local.handlers.TuyaDecoder] - 601867532462ab3a32c4/192.168.1.228:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.116 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:54.058 [ERROR] [.internal.local.handlers.TuyaDecoder] - 21410372c44f33b896bf/192.168.1.227:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.119 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:54.934 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.236', deviceId='eb349b6012eb05c13103yh', active=2, ability=0, mode=0, encrypt=true, productKey='keydqydsydtu759x', version='3.3'}'}
2022-02-18 09:22:02.120 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:55.016 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb349b6012eb05c13103yh/192.168.1.236:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.121 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newWarningEvent triggered 2022-02-18 09:21:55.233 [WARN ] [rnal.local.handlers.UserEventHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: class io.netty.handler.codec.DecoderException caught, closing the channel
2022-02-18 09:22:02.123 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:55.422 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.208', deviceId='eb04ec86abf62e08b1932f', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:02.125 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:55.436 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb3c82adf880a32bbe5ivy/192.168.1.180:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.126 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:56.709 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.148', deviceId='ebf3cb07f3574a9688z4oy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:02.127 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:57.684 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.246', deviceId='eb755c84e755bbd761ymwj', active=2, ability=0, mode=0, encrypt=true, productKey='keyyj3fy8x98arty', version='3.3'}'}
2022-02-18 09:22:02.129 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:58.205 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.226', deviceId='ebb442e79ea9c02fb35xs8', active=2, ability=0, mode=0, encrypt=true, productKey='keytg5kq8gvkv9dh', version='3.3'}'}
2022-02-18 09:22:02.130 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:58.298 [ERROR] [.internal.local.handlers.TuyaDecoder] - ebb442e79ea9c02fb35xs8/192.168.1.226:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.131 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:58.910 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb04ec86abf62e08b1932f/192.168.1.208:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:02.133 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:59.055 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.180', deviceId='eb3c82adf880a32bbe5ivy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:02.134 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:21:59.936 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.236', deviceId='eb349b6012eb05c13103yh', active=2, ability=0, mode=0, encrypt=true, productKey='keydqydsydtu759x', version='3.3'}'}
2022-02-18 09:22:02.136 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newWarningEvent triggered 2022-02-18 09:22:00.863 [WARN ] [rnal.local.handlers.UserEventHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: class io.netty.handler.codec.DecoderException caught, closing the channel
2022-02-18 09:22:02.137 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:22:00.984 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.208', deviceId='eb04ec86abf62e08b1932f', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:02.139 [INFO ] [openhab.event.ChannelTriggeredEvent ] - logreader:reader:LogReaderOpenHABCloudConnectStatus:newErrorEvent triggered 2022-02-18 09:22:01.894 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.148', deviceId='ebf3cb07f3574a9688z4oy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
==> /var/log/openhab/openhab.log <==
2022-02-18 09:22:02.684 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.246', deviceId='eb755c84e755bbd761ymwj', active=2, ability=0, mode=0, encrypt=true, productKey='keyyj3fy8x98arty', version='3.3'}'}
2022-02-18 09:22:03.207 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.226', deviceId='ebb442e79ea9c02fb35xs8', active=2, ability=0, mode=0, encrypt=true, productKey='keytg5kq8gvkv9dh', version='3.3'}'}
2022-02-18 09:22:03.705 [ERROR] [.internal.local.handlers.TuyaDecoder] - 36802558840d8e95ef5d/192.168.1.244:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:03.730 [ERROR] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a3734e6/192.168.1.229:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:03.899 [ERROR] [.internal.local.handlers.TuyaDecoder] - 42434288807d3a372fd5/192.168.1.245:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:04.065 [ERROR] [.internal.local.handlers.TuyaDecoder] - 601867532462ab3a32c4/192.168.1.228:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:04.066 [ERROR] [.internal.local.handlers.TuyaDecoder] - 21410372c44f33b896bf/192.168.1.227:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:04.198 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.180', deviceId='eb3c82adf880a32bbe5ivy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:04.936 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.236', deviceId='eb349b6012eb05c13103yh', active=2, ability=0, mode=0, encrypt=true, productKey='keydqydsydtu759x', version='3.3'}'}
2022-02-18 09:22:05.051 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb349b6012eb05c13103yh/192.168.1.236:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:05.466 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb3c82adf880a32bbe5ivy/192.168.1.180:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:05.986 [WARN ] [rnal.local.handlers.UserEventHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: class io.netty.handler.codec.DecoderException caught, closing the channel
io.netty.handler.codec.DecoderException: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:477) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.68.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.68.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.68.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.68.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [bundleFile:4.1.68.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [bundleFile:4.1.68.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719) [bundleFile:4.1.68.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655) [bundleFile:4.1.68.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581) [bundleFile:4.1.68.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) [bundleFile:4.1.68.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [bundleFile:4.1.68.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [bundleFile:4.1.68.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [bundleFile:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386) ~[?:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
==> /var/log/openhab/events.log <==
2022-02-18 09:22:05.980 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tuya:tuyaDevice:ebf3cb07f3574a9688z4oy' changed from OFFLINE to ONLINE
2022-02-18 09:22:05.997 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tuya:tuyaDevice:ebf3cb07f3574a9688z4oy' changed from ONLINE to OFFLINE
==> /var/log/openhab/openhab.log <==
2022-02-18 09:22:06.651 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.208', deviceId='eb04ec86abf62e08b1932f', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:07.050 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.148', deviceId='ebf3cb07f3574a9688z4oy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:07.684 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.246', deviceId='eb755c84e755bbd761ymwj', active=2, ability=0, mode=0, encrypt=true, productKey='keyyj3fy8x98arty', version='3.3'}'}
2022-02-18 09:22:08.207 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.226', deviceId='ebb442e79ea9c02fb35xs8', active=2, ability=0, mode=0, encrypt=true, productKey='keytg5kq8gvkv9dh', version='3.3'}'}
2022-02-18 09:22:08.331 [ERROR] [.internal.local.handlers.TuyaDecoder] - ebb442e79ea9c02fb35xs8/192.168.1.226:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:08.948 [ERROR] [.internal.local.handlers.TuyaDecoder] - eb04ec86abf62e08b1932f/192.168.1.208:6668: Received MessageWrapper{commandType=HEART_BEAT, content=''}
2022-02-18 09:22:09.280 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.180', deviceId='eb3c82adf880a32bbe5ivy', active=2, ability=0, mode=0, encrypt=true, productKey='keygg897krt5kk4t', version='3.3'}'}
2022-02-18 09:22:09.937 [ERROR] [.internal.local.handlers.TuyaDecoder] - udpListener: Received MessageWrapper{commandType=UDP_NEW, content='DiscoveryMessage{ip='192.168.1.236', deviceId='eb349b6012eb05c13103yh', active=2, ability=0, mode=0, encrypt=true, productKey='keydqydsydtu759x', version='3.3'}'}`

I tried it and the commands are now working, at least on a couple of the devices that I tried.  I will check the remainder when I have time later.

It may be related to logging for your developmental purposes, but I do see this in the log:
`2022-02-21 14:55:34.705 [WARN ] [rnal.local.handlers.UserEventHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: class io.netty.handler.codec.DecoderException caught, closing the channel
io.netty.handler.codec.DecoderException: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:477) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.72.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.72.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [bundleFile:4.1.72.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [bundleFile:4.1.72.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [bundleFile:4.1.72.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [bundleFile:4.1.72.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386) ~[?:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
`

Sure.  It is a device that shows as offline.  I disabled the Thing and the error stopped in the log.  


`2022-02-21 16:44:29.884 [DEBUG] [.internal.local.handlers.TuyaEncoder] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: Sending DP_QUERY, payload {devId=ebf3cb07f3574a9688z4oy, uid=ebf3cb07f3574a9688z4oy, t=1645483469, dps=null, gwId=ebf3cb07f3574a9688z4oy}
2022-02-21 16:44:29.886 [TRACE] [.internal.local.handlers.TuyaEncoder] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: Sending encoded '000055AA000000010000000A00000088AEBA5950965BC2EC9CD11FC6CF23BEB05FBCEEDB42C2FAA132619EAD89ADFB20B66FB78D4BC68658C93254134121DE5EF089BDD224EC53F1E310524C3D84C82C0B2123721512BBEAB032CE50A3ED74E46B8956D6462832AE542980CAE47451ABDF1FC976AE8F9003D418274707BCF1E37DE334016CFAD800FC1D4F9292D87AA85C8FB21A0000AA55'
2022-02-21 16:44:29.889 [DEBUG] [al.local.handlers.TuyaMessageHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: Connection established.
2022-02-21 16:44:29.896 [TRACE] [.internal.local.handlers.TuyaDecoder] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: Received encoded '000055AA000000010000000A0000002C00000001F08A058CF5A015BFE0E127D4B796C9882D36BADE6DB6A71DBCCDB78A4BC51EDD5132B3260000AA55'
2022-02-21 16:44:29.898 [TRACE] [.internal.local.handlers.TuyaDecoder] - ebf3cb07f3574a9688z4oy//192.168.1.148:6668: Decoded raw payload: json obj data unvalid
2022-02-21 16:44:29.901 [WARN ] [rnal.local.handlers.UserEventHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: class io.netty.handler.codec.DecoderException caught, closing the channel
io.netty.handler.codec.DecoderException: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:477) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.72.Final]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [bundleFile:4.1.72.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [bundleFile:4.1.72.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [bundleFile:4.1.72.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584) [bundleFile:4.1.72.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [bundleFile:4.1.72.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [bundleFile:4.1.72.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [bundleFile:4.1.72.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [bundleFile:4.1.72.Final]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386) ~[?:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at org.smarthomej.binding.tuya.internal.local.handlers.TuyaDecoder.decode(TuyaDecoder.java:153) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446) ~[?:?]
	... 21 more
2022-02-21 16:44:29.910 [DEBUG] [al.local.handlers.TuyaMessageHandler] - ebf3cb07f3574a9688z4oy/192.168.1.148:6668: Connection terminated.`

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/handler/TuyaDeviceHandler.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/handler/TuyaDeviceHandler.java
@@ -106,6 +106,20 @@ public class TuyaDeviceHandler extends BaseThingHandler implements DeviceInfoSub
     public void processDeviceStatus(Map<Integer, Object> deviceStatus) {
         logger.trace("'{}' received status message '{}'", thing.getUID(), deviceStatus);
 
+        if (deviceStatus.isEmpty()) {
+            // if status is empty -> need to use control method to request device status
+            Map<Integer, @Nullable Object> commandRequest = new HashMap<>();
+            dpToChannelId.keySet().forEach(dp -> commandRequest.put(dp, null));
+            dp2ToChannelId.keySet().forEach(dp -> commandRequest.put(dp, null));
+
+            TuyaDevice tuyaDevice = this.tuyaDevice;
+            if (tuyaDevice != null) {
+                tuyaDevice.set(commandRequest);
+            }
+
+            return;
+        }
+
         deviceStatus.forEach((dp, value) -> {
             String channelId = dpToChannelId.get(dp);
             if (channelId != null) {
@@ -158,7 +172,7 @@ public class TuyaDeviceHandler extends BaseThingHandler implements DeviceInfoSub
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        Map<Integer, Object> commandRequest = new HashMap<>();
+        Map<Integer, @Nullable Object> commandRequest = new HashMap<>();
 
         ChannelTypeUID channelTypeUID = channelIdToChannelTypeUID.get(channelUID.getId());
         ChannelConfiguration configuration = channelIdToConfiguration.get(channelUID.getId());

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/CommandType.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/CommandType.java
@@ -55,7 +55,8 @@ public enum CommandType {
     LAN_REMOVE_GW(249),
     LAN_CHECK_GW_UPDATE(250),
     LAN_GW_UPDATE(251),
-    LAN_SET_GW_CHANNEL(252);
+    LAN_SET_GW_CHANNEL(252),
+    DP_QUERY_NOT_SUPPORTED(-1); // this is an internal value
 
     private final int code;
 

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/TuyaDevice.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/TuyaDevice.java
@@ -93,7 +93,7 @@ public class TuyaDevice implements ChannelFutureListener {
         }
     }
 
-    public void set(Map<Integer, Object> command) {
+    public void set(Map<Integer, @Nullable Object> command) {
         MessageWrapper<?> m = new MessageWrapper<>(CommandType.CONTROL, Map.of("dps", command));
         Channel channel = this.channel;
         if (channel != null) {

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaDecoder.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaDecoder.java
@@ -32,6 +32,7 @@ import org.smarthomej.binding.tuya.internal.local.dto.TcpPayload;
 import org.smarthomej.binding.tuya.internal.util.CryptoUtil;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
 import com.google.gson.reflect.TypeToken;
 
 import io.netty.buffer.ByteBuf;
@@ -49,6 +50,8 @@ import io.netty.handler.codec.ByteToMessageDecoder;
 public class TuyaDecoder extends ByteToMessageDecoder {
     private static final Type INTEGER_OBJECT_MAP_TYPE = TypeToken
             .getParameterized(Map.class, Integer.class, Object.class).getType();
+    private static final Type STATUS_PAYLOAD_TYPE = TypeToken
+            .getParameterized(TcpPayload.class, INTEGER_OBJECT_MAP_TYPE).getType();
 
     private final Logger logger = LoggerFactory.getLogger(TuyaDecoder.class);
 
@@ -147,19 +150,27 @@ public class TuyaDecoder extends ByteToMessageDecoder {
             logger.trace("{}/{}: Decoded raw payload: {}", deviceId,
                     Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""), decodedMessage);
 
-            if (CommandType.DP_QUERY.equals(commandType) && "json obj data unvalid".equals(decodedMessage)) {
-                logger.info("{}/{}: DP_QUERY not supported. Trying to request with CONTROL.", deviceId,
-                        Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""));
-                m = new MessageWrapper<>(CommandType.DP_QUERY_NOT_SUPPORTED, Map.of());
-            } else if (CommandType.STATUS.equals(commandType) || CommandType.DP_QUERY.equals(commandType)) {
-                Type type = TypeToken.getParameterized(TcpPayload.class, INTEGER_OBJECT_MAP_TYPE).getType();
-                m = new MessageWrapper<>(commandType,
-                        Objects.requireNonNull((TcpPayload<?>) gson.fromJson(decodedMessage, type)).dps);
-            } else if (CommandType.UDP_NEW.equals(commandType)) {
-                m = new MessageWrapper<>(commandType,
-                        Objects.requireNonNull(gson.fromJson(decodedMessage, DiscoveryMessage.class)));
-            } else {
-                m = new MessageWrapper<>(commandType, decodedMessage);
+            try {
+                if (CommandType.DP_QUERY.equals(commandType) && "json obj data unvalid".equals(decodedMessage)) {
+                    // "json obj data unvalid" would also result in a JSONSyntaxException but is a known error when
+                    // DP_QUERY is not supported by the device. Using a CONTROL message with null values is a known
+                    // workaround, cf. https://github.com/codetheweb/tuyapi/blob/master/index.js#L156
+                    logger.info("{}/{}: DP_QUERY not supported. Trying to request with CONTROL.", deviceId,
+                            Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""));
+                    m = new MessageWrapper<>(CommandType.DP_QUERY_NOT_SUPPORTED, Map.of());
+                } else if (CommandType.STATUS.equals(commandType) || CommandType.DP_QUERY.equals(commandType)) {
+                    m = new MessageWrapper<>(commandType, Objects
+                            .requireNonNull((TcpPayload<?>) gson.fromJson(decodedMessage, STATUS_PAYLOAD_TYPE)).dps);
+                } else if (CommandType.UDP_NEW.equals(commandType)) {
+                    m = new MessageWrapper<>(commandType,
+                            Objects.requireNonNull(gson.fromJson(decodedMessage, DiscoveryMessage.class)));
+                } else {
+                    m = new MessageWrapper<>(commandType, decodedMessage);
+                }
+            } catch (JsonSyntaxException e) {
+                logger.warn("{}/{} failed to parse JSON: {}", deviceId,
+                        Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""), e.getMessage());
+                return;
             }
         }
 

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaDecoder.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaDecoder.java
@@ -147,7 +147,11 @@ public class TuyaDecoder extends ByteToMessageDecoder {
             logger.trace("{}/{}: Decoded raw payload: {}", deviceId,
                     Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""), decodedMessage);
 
-            if (CommandType.STATUS.equals(commandType) || CommandType.DP_QUERY.equals(commandType)) {
+            if (CommandType.DP_QUERY.equals(commandType) && "json obj data unvalid".equals(decodedMessage)) {
+                logger.info("{}/{}: DP_QUERY not supported. Trying to request with CONTROL.", deviceId,
+                        Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""));
+                m = new MessageWrapper<>(CommandType.DP_QUERY_NOT_SUPPORTED, Map.of());
+            } else if (CommandType.STATUS.equals(commandType) || CommandType.DP_QUERY.equals(commandType)) {
                 Type type = TypeToken.getParameterized(TcpPayload.class, INTEGER_OBJECT_MAP_TYPE).getType();
                 m = new MessageWrapper<>(commandType,
                         Objects.requireNonNull((TcpPayload<?>) gson.fromJson(decodedMessage, type)).dps);
@@ -159,7 +163,7 @@ public class TuyaDecoder extends ByteToMessageDecoder {
             }
         }
 
-        logger.debug("{}{}: Received {}", deviceId, Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""), m);
+        logger.debug("{}/{}: Received {}", deviceId, Objects.requireNonNullElse(ctx.channel().remoteAddress(), ""), m);
         out.add(m);
     }
 }

--- a/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaMessageHandler.java
+++ b/bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/local/handlers/TuyaMessageHandler.java
@@ -64,9 +64,11 @@ public class TuyaMessageHandler extends ChannelDuplexHandler {
             if (CommandType.DP_QUERY.equals(m.commandType) || CommandType.STATUS.equals(m.commandType)) {
                 @SuppressWarnings("unchecked")
                 Map<Integer, Object> stateMap = (Map<Integer, Object>) m.content;
-                if (stateMap != null) {
+                if (stateMap != null && !stateMap.isEmpty()) {
                     deviceStatusListener.processDeviceStatus(stateMap);
                 }
+            } else if (CommandType.DP_QUERY_NOT_SUPPORTED.equals(m.commandType)) {
+                deviceStatusListener.processDeviceStatus(Map.of());
             }
         }
     }

